ext/opcache/jit/zend_jit_x86.dasc

```diff
@@ -4609,9 +4626,8 @@ static int zend_jit_assign_dim(dasm_State **Dst, const zend_op *opline, const ze
 	}
 
 	if (op1_info & MAY_BE_REF) {
-		|	LOAD_ZVAL_ADDR FCARG1a, op1_addr
-		|	ZVAL_DEREF FCARG1a, op1_info
-		op1_addr = ZEND_ADDR_MEM_ZVAL(ZREG_FCARG1a, 0);
+		/* Needs support for typed references */
+		goto fallback;
```

Comment:
This can be implemented, but I'm not sure it makes sense. MAY_BE_REF here means no type info, so we have to emit all the checks, which will be even larger with proper support for typed references.

Reply:
better for typed references only, like you do with zend_jit_assign_op_to_typed_ref()